doc/reference/tags.md

```diff
@@ -52,3 +55,84 @@ class MyComponent extends Component {
     ...
 }
 ```
+
+## CSS tag
+
+The CSS tag is useful to define a css stylesheet in the javascript file:
+
+```js
+class MyComponent extends Component {
+    static template = xml`
+        <div class="my-component">some template</div>
+    `;
+    static css`
+      .my-component {
+          color: red;
+      }
+    `;
+}
+```
+
+The `css` tag registers internally the css information. Then, whenever the first
+instance of the component is created, will add a `<style>` tag to the document
+`<head>`.
+
+Note that to make it more useful, like other css preprocessors, the `css` tag
+accepts a small extension of the css specification: css scopes can be nested,
+and the rules will then be expanded by the `css` helper:
+
+```scss
+.my-component {
+  display: block;
+  .sub-component h {
+    color: red;
+  }
+}
+```
+
+will be formatted as:
+
+```css
+.my-component {
+  display: block;
+}
+.my-component .sub-component h {
+  color: red;
+}
+```
+
+Now, there is no additional processing done by the `css` tag. However, since it
+is done in javascript at runtime, we actually have more power. For example:
+
+1. sharing values between javascript and css:
+
+```js
+import { theme } from "./theme";
+
+class MyComponent extends Component {
+  static template = xml`<div class="my-component">...</div>`;
+  static style = css`
+    .my-component {
+      color: ${theme.MAIN_COLOR};
+      background-color: ${theme.SECONDARY_color};
+    }
+  `;
+}
+```
+
+2. scoping rules to the current component:
```

Comment:
the same would be achieved with a classname or id right?

Reply:
yes, but not exactly.  Generating a unique id guarantees that you do not clash with the same classname somewhere else in your application.  It may not be a big issue if you write an application, but for a library, it is a more important issue.  Imagine a generic spreadsheet component written in an owl library.  If this component uses a classname such a `toolbar`, or `icon`,  or whatever, there is a big chance for a conflict.  So, we need a longer and maybe not so natural classname, such as `owl-spreadsheet-toolbar`.

But yeah, that is not really a big issue.

Reply:
mmmmh good point